### PR TITLE
Migration cleanup

### LINKED ISSUE #734: Migration Cleanup
In-VS migration should clean up the on-disk .xproj and project.json files that are no longer required by the project after they've been migrated. We can't do this until after all projects have been migrated, as we migrate one project at a time and this can cause ordering issues with transitive dependencies. We need to maintain a list of all migrated projects, listen for solution load in CPS, and delete the no-longer needed files after load.

### PR DESCRIPTION
**Pull request by 333fred**
This fixes #734. We now delete the old xproj file, the xproj.user file (if it exists), the project.json, and the project.lock.json (if it exists). We also back up the xproj.user file if it exists, which brings us to feature parity with the CLI. I've made this PR depend on #1061 as they both touch the migrator factory, and it would be a royal pain to merge any other way. This will depend on a CLI insertion, or we can potentially break multi-project solutions. I've tested this with version preview5-004382 of the CLI, using a solution that previously broke if the xproj migration deleted the old files. Tagging @dotnet/project-system for review.

### CODE REVIEW

**Comment by 333fred** [comment]
@srivatsn I merged the changes from master, so now there's just the one commit's worth of changes.

**Comment by srivatsn** [comment]
I've already looked at that one commit and signed off- advertise to @dotnet/project-system so that someone else will sign off :)

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs [hunk 1]
```diff
@@ -197,6 +210,23 @@ internal bool BackupProject(string backupLocation, string xprojLocation, string
             return (report.OutputMSBuildProject, report.Succeeded);
         }
 
+        internal void CleanupXproj(string projectLocation, string projectName)
+        {
+            // Clean up the .xproj, the project.json, the project.lock.json, and the xproj.user
+            _fileSystem.RemoveFile(Path.Combine(projectLocation, $"{projectName}.xproj"));
```
Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs]: Question: Why catch exceptions for removing lock.json and .user but not the others?

Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs]: Because the others have to exist: we've already proved that. The lock file and .user files don't have to exist, so if an IOException is thrown while removing them we don't care.

Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs]: I switched these to `FileNotFoundException`s so we don't suppress unrelated file IO exceptions.
